File: php/features.php
<?php
declare(strict_types=1);

// KiprioApiSuite SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class KiprioApiSuiteFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new KiprioApiSuiteBaseFeature();
            case "test":
                return new KiprioApiSuiteTestFeature();
            default:
                return new KiprioApiSuiteBaseFeature();
        }
    }
}

File: py/features.py
# KiprioApiSuite SDK feature factory

from feature.base_feature import KiprioApiSuiteBaseFeature
from feature.test_feature import KiprioApiSuiteTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KiprioApiSuiteBaseFeature(),
        "test": lambda: KiprioApiSuiteTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
